x/merkledb/db.go

```diff
@@ -892,6 +888,10 @@ func (db *merkleDB) commitChanges(ctx context.Context, trieToCommit *trieView) e
 		return nil
 	case trieToCommit.isInvalid():
 		return ErrInvalid
+	case trieToCommit.committed:
+		return ErrCommitted
+	case trieToCommit.db != trieToCommit.getParentTrie():
```

Comment:
might be worth doing this in the trieview so we don't have to take the lock

Reply:
Personally, I feel like the readability gains from having all the checks in one place outweighs the additional amount of time we hold the lock here but I can change it if you feel strongly